### DISCUSSION TITLE
Limit Minecart Stacks

### DISCUSSION THREAD

**Discussion by IceWaffles** [PurpurMC/Purpur#1349] [Feature Requests]
Consider limiting clumped-up stacks of minecarts to groups of 10. This behavior prevents players from stacking thousands of minecarts together to create lag using collision mechanics to form a lag machine. Would be worth putting as a configuration option.

**Comment by CatTeaA** [comment]
Upstream Paper provides an option that should help you
<paper-world configuration>:
entity-per-chunk-save-limit:
    area_effect_cloud: 8
    arrow: 16
    dragon_fireball: 3
    egg: 8
    ender_pearl: 8
    experience_bottle: 3
    experience_orb: 16
    eye_of_ender: 8
    fireball: 8
    firework_rocket: 8
    llama_spit: 3
    potion: 8
    shulker_bullet: 8
    small_fireball: 8
    snowball: 8
    spectral_arrow: 16
    trident: 16
    wither_skull: 4
With the help of this entry you can set limits to how many entities of specified type can be saved. You should provide a limit for each projectile at least to avoid issues with massive amounts of projectiles being saved and your server crashing on loading that. You can put any entity id here, see the minecraft wiki to find IDs of entities. Please adjust the limit to your liking. 
Suggested value for all projectiles is around . You can also add other entities by their type names to that list. This config option is not designed to prevent players from making large mob farms.
Source：
https://github.com/YouHaveTrouble/minecraft-optimization
You can limit the number of minecar in a single block, as well as other entities, and excess entities will not be saved by the map

**Reply by IceWaffles** [reply]
This is not good enough: entity per chunk save limits is a save limit; the lag via minecarts can easily be done as long as a player stays in that area/don’t exit. Of course, once they leave, the limit gets imposed.
